src/main/java/org/onedatashare/transferservice/odstransferservice/service/step/vfs/VfsWriter.java

```diff
@@ -97,29 +102,39 @@ public void prepareFile() {
         }
     }
 
-    @BeforeRead
-    public void beforeRead() {
-        this.readStartTime = LocalDateTime.now();
-        logger.info("Before read start time {}", this.readStartTime);
+    @BeforeWrite
+    public void beforeWrite() {
+        this.writeStartTime = LocalDateTime.now();
+        logger.info("Before write start time {}", this.writeStartTime);
     }
 
-
     @Override
     public void write(List<? extends DataChunk> items) throws Exception {
         this.fileName = items.get(0).getFileName();
         this.filePath = Paths.get(this.filePath.toString(), this.fileName);
-        for (int i = 0; i < items.size(); i++) {
-            DataChunk chunk = items.get(i);
+        for (DataChunk chunk : items) {
             FileChannel channel = getChannel(chunk.getFileName());
             int bytesWritten = channel.write(ByteBuffer.wrap(chunk.getData()), chunk.getStartPosition());
-            logger.info("Wrote the amount of bytes: " + String.valueOf(bytesWritten));
+            logger.info("Wrote the amount of bytes: " + bytesWritten);
             if (chunk.getSize() != bytesWritten)
                 logger.info("Wrote " + bytesWritten + " but we should have written " + chunk.getSize());
         }
     }
 
     @AfterWrite
     public void afterWrite(List<? extends DataChunk> items) {
-        ODSConstants.metricsForOptimizerAndInflux(items, this.readStartTime, logger, stepExecution, metricCache, metricsCollector);
+        ODSConstants.metricsForOptimizerAndInflux(items, this.writeStartTime, logger, stepExecution, metricCache, metricsCollector);
+    }
+
+
+    @BeforeRead
+    public void beforeRead() {
```

Comment:
Since we are now keeping track of reading adn writing separately, what you can do is embed this monitoring logic like in VFS Writer in the readers as well.